### ISSUE TITLE
Out of memory exception

### ISSUE THREAD

**Issue by martinrotter** [codebude/QRCoder#207]
## Type of issue
[ x] Bug  
[ ] Question (e.g. about handling/usage)  
[ ] Request for new feature/improvement  

## Expected Behavior

Returns QR code, which is then printed as graphic object.

## Current Behavior

Crash with out-of-memory exception.

## Steps to Reproduce (for bugs)

https://paste.ubuntu.com/p/CFPTf4JMt7/

## Your Environment
<!--
Include as many relevant details about the environment you experienced the bug in.  
Windows 7.

* Version used:  latest master, as of 15.10.2020, but same thing happened with 1.3.3
* Compiled from source or NuGet package?:  From source
* Payload/Encoded text:  happens probably with any text

```text
 ===== VÝJIMKA =====
System.OutOfMemoryException: Byla vyvolána výjimka typu System.OutOfMemoryException.
   v System.Linq.Lookup`2.GetGrouping(TKey key, Boolean create)
   v System.Linq.Lookup`2.Create[TSource](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
   v System.Linq.GroupedEnumerable`3.GetEnumerator()
   v System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   v System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   v System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   v QRCoder.QRCodeGenerator.BinaryStringToBitBlockList(String bitString) v d:\ServerFolders\SpravaProjektu\Jenkins\workspace\QRCoder\QRCoder\QRCodeGenerator.cs:řádek 935
   v QRCoder.QRCodeGenerator.GenerateQrCode(String bitString, ECCLevel eccLevel, Int32 version) v d:\ServerFolders\SpravaProjektu\Jenkins\workspace\QRCoder\QRCoder\QRCodeGenerator.cs:řádek 196
   v QRCoder.QRCodeGenerator.GenerateQrCode(String plainText, ECCLevel eccLevel, Boolean forceUtf8, Boolean utf8BOM, EciMode eciMode, Int32 requestedVersion) v d:\ServerFolders\SpravaProjektu\Jenkins\workspace\QRCoder\QRCoder\QRCodeGenerator.cs:řádek 145
   v QRCoder.QRCodeGenerator.CreateQrCode(String plainText, ECCLevel eccLevel, Boolean forceUtf8, Boolean utf8BOM, EciMode eciMode, Int32 requestedVersion) v d:\ServerFolders\SpravaProjektu\Jenkins\workspace\QRCoder\QRCoder\QRCodeGenerator.cs:řádek 76
   v PraktikSW.SluzbyPraktika.ModulTisk.TiskoveNastroje.TiskovaUloha.TisknoutStranku(Object sender, PrintPageEventArgs e)
   v System.Drawing.Printing.PrintDocument.OnPrintPage(PrintPageEventArgs e)
   v System.Drawing.Printing.PrintDocument._OnPrintPage(PrintPageEventArgs e)
   v System.Drawing.Printing.PrintController.PrintLoop(PrintDocument document)
   v System.Drawing.Printing.PrintController.Print(PrintDocument document)
   v System.Drawing.Printing.PrintDocument.Print()
   v PraktikSW.SluzbyPraktika.ModulTisk.TiskoveNastroje.TiskovaUloha.Vytisknout(Fronta fronta, TiskDokumentuArgumentUdalosti pozadavek, Formular formular, Boolean rozsirenyTisk, PrintDocument dokument, ModulTisk modul)
   v PraktikSW.SluzbyPraktika.ModulTisk.Sluzby.TiskDokumentu.VytisknoutUlohu(Fronta fronta, TiskovaUloha uloha, TiskDokumentuArgumentUdalosti pozadavek)
 ===== VÝJIMKA =====
```

**Comment by martinrotter** [comment]
===== VÝJIMKA =====
System.OutOfMemoryException: Byla vyvolána výjimka typu System.OutOfMemoryException.
   v System.Linq.Lookup`2.GetGrouping(TKey key, Boolean create)
   v System.Linq.Lookup`2.Create[TSource](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
   v System.Linq.GroupedEnumerable`3.GetEnumerator()
   v System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
   v System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   v System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   v QRCoder.QRCodeGenerator.BinaryStringToBitBlockList(String bitString) v d:\ServerFolders\SpravaProjektu\Jenkins\workspace\QRCoder\QRCoder\QRCodeGenerator.cs:řádek 935
   v QRCoder.QRCodeGenerator.GenerateQrCode(String bitString, ECCLevel eccLevel, Int32 version) v d:\ServerFolders\SpravaProjektu\Jenkins\workspace\QRCoder\QRCoder\QRCodeGenerator.cs:řádek 196
   v QRCoder.QRCodeGenerator.GenerateQrCode(String plainText, ECCLevel eccLevel, Boolean forceUtf8, Boolean utf8BOM, EciMode eciMode, Int32 requestedVersion) v d:\ServerFolders\SpravaProjektu\Jenkins\workspace\QRCoder\QRCoder\QRCodeGenerator.cs:řádek 145
   v QRCoder.QRCodeGenerator.CreateQrCode(String plainText, ECCLevel eccLevel, Boolean forceUtf8, Boolean utf8BOM, EciMode eciMode, Int32 requestedVersion) v d:\ServerFolders\SpravaProjektu\Jenkins\workspace\QRCoder\QRCoder\QRCodeGenerator.cs:řádek 76
   v PraktikSW.SluzbyPraktika.ModulTisk.TiskoveNastroje.TiskovaUloha.TisknoutStranku(Object sender, PrintPageEventArgs e)
   v System.Drawing.Printing.PrintDocument.OnPrintPage(PrintPageEventArgs e)
   v System.Drawing.Printing.PrintDocument._OnPrintPage(PrintPageEventArgs e)
   v System.Drawing.Printing.PrintController.PrintLoop(PrintDocument document)
   v System.Drawing.Printing.PrintController.Print(PrintDocument document)
   v System.Drawing.Printing.PrintDocument.Print()
   v PraktikSW.SluzbyPraktika.ModulTisk.TiskoveNastroje.TiskovaUloha.Vytisknout(Fronta fronta, TiskDokumentuArgumentUdalosti pozadavek, Formular formular, Boolean rozsirenyTisk, PrintDocument dokument, ModulTisk modul)
   v PraktikSW.SluzbyPraktika.ModulTisk.Sluzby.TiskDokumentu.VytisknoutUlohu(Fronta fronta, TiskovaUloha uloha, TiskDokumentuArgumentUdalosti pozadavek)
 ===== VÝJIMKA =====

**Comment by codebude** [comment]
Hi @martinrotter ,

@jnyrup did a lot of memory optimization in this merge request: https://github.com/codebude/QRCoder/pull/229

Since I never could produce an out of memory exception, I'll close the issue for new. Please re-open the issue if the error still occurs. (The changes will be available from the 1.3.7 Nuget package - or if you want to try them out right now - from the myget package feed. Check the readme.md file on how to use the myget feed.)
